content/guides/end-to-end-testing/amazon-cognito-authentication.md

```diff
@@ -157,6 +158,169 @@ const awsConfig = require('./aws-exports-es5.js')
 
 ## Custom Command for Amazon Cognito Authentication
 
+There are two ways you can authenticate to AWS Cognito
+
+- [Login with cy.origin()](/guides/end-to-end-testing/amazon-cognito-authentication#Login-with-cy-origin)
+- [Legacy Programmatic Access](/guides/end-to-end-testing/amazon-cognito-authentication#Programmatic-Login-Legacy)
+
+### Login with [`cy.origin()`](/api/commands/origin)
+
+Next, we'll write a command to perform a login to
+[Amazon Cognito](https://aws.amazon.com/cognito) using
+[`cy.origin()`](/api/commands/origin) to navigate to the cognito origin,
+inputting a user credentials, and signing in via OAuth to redirect back to our
+application, and caching the results with
+[`cy.session()`](/api/commands/session).
+
+In this `loginByCognito` command, we redirect to the cognito login screen by
+authenticating with `Sign in with AWS`. We then enter our user credentials and
+sign in, which redirects us back to the realworld app.
+
+```jsx
+// cypress/support/auth-provider-commands/cognito.ts
+
+/// Amazon Cognito
+Cypress.Commands.add('loginByCognito', (username, password) => {
+  Cypress.log({
+    displayName: 'COGNITO LOGIN',
+    message: [`🔐 Authenticating | ${username}`],
+    // @ts-ignore
+    autoEnd: false,
+  })
+
+  cy.visit('/')
+  cy.contains('Sign in with AWS', {
+    includeShadowDom: true,
```

Comment:
just curious, is this needed?

Reply:
the cognito react components are all shadow DOM roots 🥴